### PR TITLE
[AIRFLOW-5395] Fix unloading with column names containing a dot in RedshiftToS3Operator

### PR DESCRIPTION
**Pull request by feluelle**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-5395
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

Currently you cannot `include_header` when there is a dot (for example) in the header names. To allow it, we can simply enclose the names via `""`.
```diff
+ column_names = ', '.join('"{0}"'.format(c) for c in columns)
- column_names = ', '.join("{0}".format(c) for c in columns)
+ column_castings = ', '.join('CAST("{0}" AS text) AS "{0}"'.format(c)
- column_castings = ', '.join("CAST({0} AS text) AS {0}".format(c)
```

FYI: https://docs.aws.amazon.com/en_pv/redshift/latest/dg/r_names

This PR also fixes Pylint Issues and changes the instance attributes "hook" and "s3" to be local variables.

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

**Comment by feluelle** [comment]
I just saw there is a PR #4610 which adds the new `HEADER` argument and apparently the author stopped working on it. So should I implement that here instead?

**Comment by OmerJog** [comment]
@feluelle I think the two are separated issues. Maybe better to rePR #4610 separately

**Comment by feluelle** [comment]
@OmerJog that's true, but if the other one gets merged this change is superfluous.

I resolved conflicts. Can you review it since you were also interested in the other one ;)

File: airflow/operators/redshift_to_s3_operator.py [hunk 1]
```diff
@@ -106,13 +114,13 @@ def execute(self, context):
                             """.format(schema=self.schema,
                                        table=self.table)
 
-            cursor = self.hook.get_conn().cursor()
+            cursor = postgres_hook.get_conn().cursor()
             cursor.execute(columns_query)
             rows = cursor.fetchall()
             columns = [row[0] for row in rows]
-            column_names = ', '.join("{0}".format(c) for c in columns)
+            column_names = ', '.join('"{0}"'.format(c) for c in columns)
```
Comment (ashb) [inline airflow/operators/redshift_to_s3_operator.py:121]: Lets use `from psycopg2.extensions import quote_ident` to be more resilient.

```suggestion
            column_names = ', '.join(quote_ident(c, scope=cursor) in columns)
```

**Review by ashb — CHANGES_REQUESTED** [review]

**Comment by ashb** [comment]
@feluelle Thought if you want to "take-over" the other PR that might be better - open a new PR, fix the issues and annotate the commit with `Co-authored-by:`? https://github.blog/2018-01-29-commit-together-with-co-authors/

**Comment by feluelle** [comment]
> @feluelle Thought if you want to "take-over" the other PR that might be better - open a new PR, fix the issues and annotate the commit with `Co-authored-by:`? https://github.blog/2018-01-29-commit-together-with-co-authors/

Will do that - on monday when I am back at a "good" computer 😁

**Comment by feluelle** [comment]
Closed in favor of #6309

**Pull request closed without merging**